### PR TITLE
[velero]Bump velero version v1.12.0

### PR DESCRIPTION
**Pull request by qiuming-best**
#### Special notes for your reviewer:
- Bump up velero version to v1.12.0
- [Enforce Pod Security Standards with Namespace Labels](https://kubernetes.io/docs/tasks/configure-pod-container/enforce-standards-namespace-labels/):
   - by creating one job to label the Velero namespace with specific labels, which would not create a Velero namespace in case of  conflict with helm install ----create-namespace flag
#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the values.yaml or README.md
- [x] Title of the PR starts with chart name (e.g. `[velero]`)

### CODE REVIEW

**Comment by hmlkao** [comment]
Velero v1.12.0 was released, can you rerun checks?

**Comment by JeyDi** [comment]
Hi guys! There are any updates on this? We really need the helm charts :) Thank you!

**Comment by qiuming-best** [comment]
@JeyDi @hmlkao I'm working on it, but it has some git action error, we will fix it, and the v1.12.0 helm chart will release soon

**Comment by qiuming-best** [comment]
@jenting 
Here I'm encountering two errors in GitHub action:
1. Check Velero Manifest
    I've run the script locally, and the difference happened in the newly added two CRDs(defined in `dataupload.yaml` and `datadownload.yaml`) in Velero 1.12.0
2. Lint test failed to install Velero for `unable to get key for secret: Secret \"test-credential\" not found)`
    I'm not sure where the `test-credential` secret created
    
Please help me figure it out, many thanks!

**Comment by jenting** [comment]
1. Could you please help update the `.github/workflows/verify-manifest.yml` as follows?
    ```yaml
    - name: Compare manifests
        run: |
          cd charts/velero
          mkdir -p diff_workspace

          echo "Generate velero manifest file"
          velero install --crds-only --dry-run -o yaml | kubectl apply --dry-run=client -f - -o yaml > diff_workspace/velero_manifest.yaml

          echo "Generate crds manifest file"
          kubectl apply --dry-run=client -f crds/ -o yaml > diff_workspace/crds_manifest.yaml

          echo "Filter non-required values"
          for yaml_file in diff_workspace/*.yaml; do
            yq -i '.items |= sort_by(.metadata.name)' "$yaml_file"
            yq -i 'del(.items[].metadata.creationTimestamp)' "$yaml_file"
            yq -i 'del(.items[].metadata.annotations."kubectl.kubernetes.io/last-applied-configuration")' "$yaml_file"
            yq -i 'del(.status)' "$yaml_file"
          done

          echo "Compare the two files using diff"
          difference=$(diff diff_workspace/velero_manifest.yaml diff_workspace/crds_manifest.yaml) || true
          if [ -z "$difference" ]; then
            echo "Files are synced. No differences found."
          else
            echo "Differences between velero_manifest.yaml and crds_manifest.yaml:"
            echo "$difference"
            exit 1
          fi
    ```

2. I'm still figuring out the chart upgrade test failure reason. But from my observation, the error message `unable to get key for secret: Secret \"test-credential\" not found)` is not the root cause.

**Comment by qiuming-best** [comment]
> 1. Could you please help update the `.github/workflows/verify-manifest.yml` as follows?
>    ```yaml
>    - name: Compare manifests
>        run: |
>          cd charts/velero
>          mkdir -p diff_workspace
>    
>          echo "Generate velero manifest file"
>          velero install --crds-only --dry-run -o yaml | kubectl apply --dry-run=client -f - -o yaml > diff_workspace/velero_manifest.yaml
>    
>          echo "Generate crds manifest file"
>          kubectl apply --dry-run=client -f crds/ -o yaml > diff_workspace/crds_manifest.yaml
>    
>          echo "Filter non-required values"
>          for yaml_file in diff_workspace/*.yaml; do
>            yq -i '.items |= sort_by(.metadata.name)' "$yaml_file"
>            yq -i 'del(.items[].metadata.creationTimestamp)' "$yaml_file"
>            yq -i 'del(.items[].metadata.annotations."kubectl.kubernetes.io/last-applied-configuration")' "$yaml_file"
>            yq -i 'del(.status)' "$yaml_file"
>          done
>    
>          echo "Compare the two files using diff"
>          difference=$(diff diff_workspace/velero_manifest.yaml diff_workspace/crds_manifest.yaml) || true
>          if [ -z "$difference" ]; then
>            echo "Files are synced. No differences found."
>          else
>            echo "Differences between velero_manifest.yaml and crds_manifest.yaml:"
>            echo "$difference"
>            exit 1
>          fi
>    ```
> 2. I'm still figuring out the chart upgrade test failure reason. But from my observation, the error message `unable to get key for secret: Secret \"test-credential\" not found)` is not the root cause.

@jenting Thanks, updated

File: charts/velero/templates/label-namespace/labelnamespace.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          - |
+            {{- with .Values.namespace }}
+            {{- range .labels }}
+            kubectl label namespace {{ $.Release.Namespace }} {{ .key }}={{ .value }}
+            {{- end }}
+            {{- end }}
+        restartPolicy: Never
+    backoffLimit: 3
+  {{- end }}
+{{- end }}
```
Comment (jenting) [inline charts/velero/templates/label-namespace/labelnamespace.yaml:41]:
```suggestion
{{- if .Values.namespace }}
{{- if gt (len .Values.namespace.labels) 0 }}
apiVersion: batch/v1
kind: Job
metadata:
  name: {{ template "velero.fullname" . }}-label-namespace
  namespace: {{ .Release.Namespace }}
  annotations:
    "helm.sh/hook": post-install,post-upgrade,post-rollback
    "helm.sh/hook-delete-policy": hook-succeeded
  labels:
    app.kubernetes.io/name: {{ include "velero.name" . }}
    app.kubernetes.io/instance: {{ .Release.Name }}
    app.kubernetes.io/managed-by: {{ .Release.Service }}
    helm.sh/chart: {{ include "velero.chart" . }}
spec:
  template:
    spec:
      serviceAccountName: {{ include "velero.serverServiceAccount" . }}
      containers:
      - name: label-namespace
        {{- if .Values.kubectl.image.digest }}
        image: "{{ .Values.kubectl.image.repository }}@{{ .Values.kubectl.image.digest }}"
        {{- else if .Values.kubectl.image.tag }}
        image: "{{ .Values.kubectl.image.repository }}:{{ .Values.kubectl.image.tag }}"
        {{- else }}
        image: "{{ .Values.kubectl.image.repository }}:{{ template "chart.KubernetesVersion" . }}"
        {{- end }}
        command:
        - /bin/sh
        - -c
        - |
          {{- range .Values.namespace.labels }}
          kubectl label namespace {{ $.Release.Namespace }} {{ .key }}={{ .value }}
          {{- end }}
      restartPolicy: Never
  backoffLimit: 3
{{- end }}
{{- end }}
```

**Review by jenting — APPROVED** [review]

**Comment by jenting** [comment]
@qiuming-best I have addressed the issue 😀

**Comment by jack-nix** [comment]
Hello @reasonerjt @ywk253100 could you please review the MR? thanks.

**Review by ywk253100 — APPROVED** [review]

**Pull request merged**